Update `typescript` to version 5.6.2

I have tried to perform this update, but the types test fail when doing so.

Steps to reproduce:

1. Update the `typescript` dependency to version 5.6.2 in `package.json` and install it.
2. Run `npx gulp typestest`.
3. Notice that it fails with the following error: `Couldn't compile TypeScript test: build/typestest/types/src/display/display_utils.d.ts(89,19): error TS2339: Property 'docId' does not exist on type '{ ownerDocument?: Document | undefined; }'.`

The error seems to happen on line https://github.com/mozilla/pdf.js/blob/master/src/display/display_utils.js#L66, but AFAICT this is valid code.

/cc @tamuratak @ineiti Could one of you perhaps help out with this?

--- a/package.json
+++ b/package.json
@@ -54,7 +54,7 @@
     "terser-webpack-plugin": "^5.3.10",
     "tsc-alias": "^1.8.10",
     "ttest": "^4.0.0",
-    "typescript": "^5.5.4",
+    "typescript": "^5.6.2",
     "vinyl": "^3.0.0",
     "webpack": "^5.94.0",
     "webpack-stream": "^7.0.0",

